clients/go/apiban-ipsets/apiban-ipsets.go

```diff
@@ -293,11 +310,38 @@ func run(ctx context.Context, blset ipset.IPSet, apiconfig ApibanConfig) error {
 							log.Print("Entry NOT existing...")
 						}
 					*/
-					err := blset.Add(s, 0)
+					// check if the "IP" field is present
+					ip, ok := s["IP"]
+					if !ok {
+						continue
+					}
+					// check the string type for the "IP" field
+					ipStr, ok := ip.(string)
+					if !ok {
+						continue
+					}
+					// check if "encrypt" field is present
+					if kvCode, ok := s["encrypt"]; ok {
+						// check the string type for "encrypt" field
+						kvCodeStr, ok := kvCode.(string)
+						if !ok {
+							continue
+						}
+						if !encryptionOn {
+							log.Print("IP encrypted but no passphrase configured")
+							continue
+						}
```

Comment:
Also as I'm not literate in `go` -- what does exactly happen when password is configured but a password-less event is encountered -- it seems the IP address is skipped? I would vote for being liberal, issuing a warning and including it in IPSET. Motivation: we could then more easily mix up user's IP addresses with some public shared IP address list, like honeynet.